panic in data node liquidity provision observer

v0.57.0 

observer not initialised in datanode/sqlstore/liquidity_provision.go constructor. 

```
panic: assignment to entry in nil map
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: goroutine 4702 [running]:
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: code.vegaprotocol.io/vega/datanode/utils.(*Observer[...]).Subscribe(0xc000f1e548, {0x2acf220, 0xc00277e780}, 0xc0027deeb0)
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/work/vega/vega/datanode/utils/observer.go:57 +0xdc
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: code.vegaprotocol.io/vega/datanode/utils.(*Observer[...]).Observe(0xc000f1e548, {0x2acf220, 0xc00277e780}, 0x3, 0xc001501b78)
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/work/vega/vega/datanode/utils/observer.go:114 +0x7c
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: code.vegaprotocol.io/vega/datanode/sqlstore.(*LiquidityProvision).ObserveLiquidityProvisions(0xc000f1e510, {0x2acf220, 0xc00277e780}, 0xc00277e740?, 0x0, 0xc0027dee90)
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/work/vega/vega/datanode/sqlstore/liquidity_provision.go:65 +0xb1
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: code.vegaprotocol.io/vega/datanode/api.(*tradingDataServiceV2).ObserveLiquidityProvisions(0xc0012c9040, 0xc00277e740, {0x2ad8990, 0xc0027deea0})
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/work/vega/vega/datanode/api/trading_data_v2.go:1752 +0xcb
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_ObserveLiquidityProvisions_Handler({0x247b400?, 0xc0012c9040}, {0x2ad60c0, 0xc00152c1a0})
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/work/vega/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:2800 +0xd3
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: google.golang.org/grpc.(*Server).processStreamingRPC(0xc000de14a0, {0x2add100, 0xc002d0a680}, 0xc002698c60, 0xc000e6ca20, 0x3d28620, 0x0)
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1558 +0xfc7
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: google.golang.org/grpc.(*Server).handleStream(0xc000de14a0, {0x2add100, 0xc002d0a680}, 0xc002698c60, 0x0)
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]:         /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1640 +0x9ea
Oct 04 18:38:57 n07.testnet.vega.xyz visor[118425]: google.golang.org/grpc.(*Server).serveStreams.func1.2()
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,7 @@
 - [6374](https://github.com/vegaprotocol/vega/issues/6374) - Fix panic with the metrics
 - [6373](https://github.com/vegaprotocol/vega/issues/6373) - Fix panic with the metrics as well
 - [6395](https://github.com/vegaprotocol/vega/issues/6395) - Wallet selection doesn't lower case the wallet name during input verification
+- [6408](https://github.com/vegaprotocol/vega/issues/6408) - Initialise observer in liquidity provision `sql` store
 
 ## 0.57.0
 

--- a/cmd/data-node/commands/start/sqlsubscribers.go
+++ b/cmd/data-node/commands/start/sqlsubscribers.go
@@ -190,7 +190,7 @@ func (s *SQLSubscribers) CreateAllStores(ctx context.Context, Log *logging.Logge
 	s.positionStore = sqlstore.NewPositions(transactionalConnectionSource)
 	s.oracleSpecStore = sqlstore.NewOracleSpec(transactionalConnectionSource)
 	s.oracleDataStore = sqlstore.NewOracleData(transactionalConnectionSource)
-	s.liquidityProvisionStore = sqlstore.NewLiquidityProvision(transactionalConnectionSource)
+	s.liquidityProvisionStore = sqlstore.NewLiquidityProvision(transactionalConnectionSource, Log)
 	s.transfersStore = sqlstore.NewTransfers(transactionalConnectionSource)
 	s.stakeLinkingStore = sqlstore.NewStakeLinking(transactionalConnectionSource)
 	s.notaryStore = sqlstore.NewNotary(transactionalConnectionSource)

--- a/datanode/api/trading_test.go
+++ b/datanode/api/trading_test.go
@@ -135,7 +135,7 @@ func getTestGRPCServer(t *testing.T, ctx context.Context) (tidy func(), conn *gr
 	sqlPartyService := service.NewParty(sqlstore.NewParties(sqlConn), logger)
 	sqlOracleSpecService := service.NewOracleSpec(sqlstore.NewOracleSpec(sqlConn), logger)
 	sqlOracleDataService := service.NewOracleData(sqlstore.NewOracleData(sqlConn), logger)
-	sqlLPDataService := service.NewLiquidityProvision(sqlstore.NewLiquidityProvision(sqlConn), logger)
+	sqlLPDataService := service.NewLiquidityProvision(sqlstore.NewLiquidityProvision(sqlConn, logger), logger)
 	sqlTransferService := service.NewTransfer(sqlstore.NewTransfers(sqlConn), logger)
 	sqlStakeLinkingService := service.NewStakeLinking(sqlstore.NewStakeLinking(sqlConn), logger)
 	sqlNotaryService := service.NewNotary(sqlstore.NewNotary(sqlConn), logger)

--- a/datanode/sqlstore/liquidity_provision.go
+++ b/datanode/sqlstore/liquidity_provision.go
@@ -20,6 +20,7 @@ import (
 	"code.vegaprotocol.io/vega/datanode/entities"
 	"code.vegaprotocol.io/vega/datanode/metrics"
 	"code.vegaprotocol.io/vega/datanode/utils"
+	"code.vegaprotocol.io/vega/logging"
 	v2 "code.vegaprotocol.io/vega/protos/data-node/api/v2"
 	"github.com/georgysavva/scany/pgxscan"
 )
@@ -40,11 +41,12 @@ const (
 		commitment_amount, fee, sells, buys, version, status, reference, tx_hash, vega_time`
 )
 
-func NewLiquidityProvision(connectionSource *ConnectionSource) *LiquidityProvision {
+func NewLiquidityProvision(connectionSource *ConnectionSource, log *logging.Logger) *LiquidityProvision {
 	return &LiquidityProvision{
 		ConnectionSource: connectionSource,
 		batcher: NewMapBatcher[entities.LiquidityProvisionKey, entities.LiquidityProvision](
 			"liquidity_provisions", entities.LiquidityProvisionColumns),
+		observer: utils.NewObserver[entities.LiquidityProvision]("liquidity_provisions", log, 10, 10),
 	}
 }
 

--- a/datanode/sqlstore/liquidity_provision_test.go
+++ b/datanode/sqlstore/liquidity_provision_test.go
@@ -20,6 +20,7 @@ import (
 
 	"code.vegaprotocol.io/vega/datanode/entities"
 	"code.vegaprotocol.io/vega/datanode/sqlstore"
+	"code.vegaprotocol.io/vega/logging"
 	"code.vegaprotocol.io/vega/protos/vega"
 	"github.com/jackc/pgx/v4"
 	"github.com/stretchr/testify/assert"
@@ -50,7 +51,7 @@ func setupLPTests(t *testing.T, ctx context.Context) (*sqlstore.Blocks, *sqlstor
 	DeleteEverything()
 
 	bs := sqlstore.NewBlocks(connectionSource)
-	lp := sqlstore.NewLiquidityProvision(connectionSource)
+	lp := sqlstore.NewLiquidityProvision(connectionSource, logging.NewTestLogger())
 
 	config := NewTestConfig(testDBPort)
 	conn, err := pgx.Connect(ctx, config.ConnectionConfig.GetConnectionString())

